assets/scripts/Actor/Main_Character/MainCharacter.ts

```diff
@@ -4,10 +4,32 @@ import DIRECTION from '../DIRECTION';
 const { ccclass } = cc._decorator;
 @ccclass
 export default class MainCharacter extends Actor {
+  private facing: DIRECTION = DIRECTION.RIGHT;
+
   public move(direction: DIRECTION): void {
+    if (direction !== this.facing && direction !== DIRECTION.IDLE) {
+      this.refacing(direction);
+    }
     this.direction = direction;
   }
 
+  public get isMoving(): boolean {
+    return this.rigidBody.linearVelocity.x !== 0;
+  }
+
+  public jump(): void {
+    if (this.isInTheFloor) {
+      this.isJumping = true;
+      this.rigidBody.applyForceToCenter(cc.v2(0, this.jumpForce), true);
+    }
+  }
+
+  public refacing(direction: DIRECTION): void {
```

Comment:
```suggestion
  public reface(direction: DIRECTION): void {
```
pfv corrija "refacing" parece mais nome de propriedade que de função.